### ISSUE TITLE
XmlDocument::Validate() should probably validate its parameters

### ISSUE THREAD

**Issue by xrmx** [marudor/libxmljs2#6]
If we pass it a Node::buffer node will segfault:

```
0x00007ffff12e1d4d in libxmljs::XmlDocument::Validate(Nan::FunctionCallbackInfo<v8::Value> const&) ()
   from /home/rm/node_modules/libxmljs2/build/Release/xmljs.node
(gdb) bt full
#0  0x00007ffff12e1d4d in libxmljs::XmlDocument::Validate(Nan::FunctionCallbackInfo<v8::Value> const&)
    () from /home/rm/node_modules/libxmljs2/build/Release/xmljs.node
No symbol table info available.
#1  0x00007ffff12e0d27 in Nan::imp::FunctionCallbackWrapper ()
   from /home/rm/node_modules/libxmljs2/build/Release/xmljs.node
```

If we pass it a string node will sigabrt:

```
node: ../node_modules/nan/nan_object_wrap.h:32: static T* Nan::ObjectWrap::Unwrap(v8::Local<v8::Object>) [with T = libxmljs::XmlDocument]: Assertion `object->InternalFieldCount() > 0' failed.

Thread 1 "node" received signal SIGABRT, Aborted.
__GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
50	../sysdeps/unix/sysv/linux/raise.c: File o directory non esistente.
(gdb) bt full
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
        set = {__val = {0, 5832576, 140737488326896, 140737286014071, 10196179845120, 5832576, 5832677, 
            5832576, 5832576, 5832768, 5832876, 5832576, 5832876, 0, 0, 0}}
        pid = <optimized out>
        tid = <optimized out>
        ret = <optimized out>
#1  0x00007ffff3eb2535 in __GI_abort () at abort.c:79
        save_stage = 1
        act = {__sigaction_handler = {sa_handler = 0x58ff80, sa_sigaction = 0x58ff80}, sa_mask = {
            __val = {0, 140737287339072, 5832784, 0, 0, 0, 21474836480, 140737488326648, 
              140737488326496, 140737287129776, 140737287114464, 0, 2807423946951993856, 
              140737287099322, 0, 140737287114464}}, sa_flags = -247420968, 
          sa_restorer = 0x7ffff140a800}
        sigs = {__val = {32, 0 <repeats 15 times>}}
#2  0x00007ffff3eb240f in __assert_fail_base (
    fmt=0x7ffff4014ee0 "%s%s%s:%u: %s%sAssertion `%s' failed.\n%n", 
    assertion=0x7ffff140a800 "object->InternalFieldCount() > 0", 
    file=0x7ffff140a7d8 "../node_modules/nan/nan_object_wrap.h", line=32, function=<optimized out>)
    at assert.c:92
        str = 0x58ff80 "\320*P"
        total = 4096
#3  0x00007ffff3ec0102 in __GI___assert_fail (
    assertion=0x7ffff140a800 "object->InternalFieldCount() > 0", 
    file=0x7ffff140a7d8 "../node_modules/nan/nan_object_wrap.h", line=32, 
    function=0x7ffff140b260 <Nan::ObjectWrap::Unwrap<libxmljs::XmlDocument>(v8::Local<v8::Object>)::__PRETTY_FUNCTION__> "static T* Nan::ObjectWrap::Unwrap(v8::Local<v8::Object>) [with T = libxmljs::XmlDocument]") at assert.c:101
```

**Comment by marudor** [comment]
Fixed by v0.21.4

**Comment by xrmx** [comment]
Thank you!
